mephisto/data_model/blueprint.py

```diff
@@ -38,17 +40,50 @@
     from argparse import _ArgumentGroup as ArgumentGroup
 
 
+@dataclass
+class BlueprintArgs:
+    _blueprint_type: str = MISSING
+    onboarding_qualification: str = field(
+        default=MISSING,
+        metadata={
+            'help': (
+                "Specify the name of a qualification used to block workers who fail onboarding, "
+                "Empty will skip onboarding."
+            ),
+        },
+    )
+    block_qualification: str = field(
+        default=MISSING,
+        metadata={
+            'help': (
+                "Specify the name of a qualification used to soft block workers."
+            ),
+        },
+    )
+
+
+@dataclass
+class SharedTaskState:
```

Comment:
I sort of prefer Data over State here only because this seems like something that doesn't necessarily change over the lifetime of the program, whereas I expect state to be something that updates. I could be wrong with this assumption though.

Reply:
State will be shared between every running task available to the `TaskRunner`s. At the moment we don't have any demo tasks that use this for the purpose of sharing information between tasks, but it's something that is actively used elsewhere. It also is a bag to receive additional 'shared' context like functions and other things that can't be passed via configuration (for instance, a Stack containing a bunch of things, or an object that generates new tasks based on time of day, etc). 

Any idea what might be a better name for this?

Reply:
I didn't realize the use case you just mentioned, I was just going off of this example. Given that understanding, I do like the name more now. Thanks for explaining!